integration_tests/src/testcases/locator.test.ts

```diff
@@ -0,0 +1,455 @@
+      codeIds.factory,
+      factory2_instantiate_message,
+      'drop-staking-factory',
+      'auto',
+      [],
+    );
+    expect(instantiate1Res.contractAddress).toHaveLength(66);
+    expect(instantiate2Res.contractAddress).toHaveLength(66);
+    expect(instantiate1Res.contractAddress).not.toBe(
+      instantiate2Res.contractAddress,
+    );
+    context.contracts.factories = [
+      new DropFactory.Client(client, instantiate1Res.contractAddress),
+      new DropFactory.Client(client, instantiate2Res.contractAddress),
+    ];
+  });
+  it('Instantiate locator contract', async () => {
+    const { client, account, codeIds } = context;
+    const instantiateRes = await DropLocator.Client.instantiate(
+      client,
+      account.address,
+      codeIds.locator,
+      {},
+      'drop-locator',
+      'auto',
+      [],
+    );
+    expect(instantiateRes.contractAddress).toHaveLength(66);
+    context.contracts.locator = new DropLocator.Client(
+      client,
+      instantiateRes.contractAddress,
+    );
+  });
+  it('Add factory instances to locator contract & Make sure they were added', async () => {
+    const { locator, factories } = context.contracts;
+    const { account } = context;
+    await locator.addChains(
+      account.address,
+      {
+        chains: [
+          {
+            name: 'instance_1',
+            factory_addr: factories[0].contractAddress,
+          },
+          {
+            name: 'instance_2',
+            factory_addr: factories[1].contractAddress,
+          },
+        ],
+      },
+      'auto',
+      '',
+      [],
+    );
+    expect((await locator.queryChains()).length).toEqual(2);
+    expect((await locator.queryFactoryInstances()).length).toEqual(2);
+  });
+  it('Query factory instances', async () => {
+    const { locator } = context.contracts;
+    const res = await locator.queryFactoryInstances();
+    expect(res).toHaveLength(2);
+    for (const instance of res) {
+      expect(instance.addr).toHaveLength(66);
+      expect(instance.contracts.token_contract).toHaveLength(66);
+      expect(instance.contracts.core_contract).toHaveLength(66);
+      expect(instance.contracts.puppeteer_contract).toHaveLength(66);
+      expect(instance.contracts.staker_contract).toHaveLength(66);
+      expect(instance.contracts.withdrawal_voucher_contract).toHaveLength(66);
+      expect(instance.contracts.withdrawal_manager_contract).toHaveLength(66);
+      expect(instance.contracts.strategy_contract).toHaveLength(66);
+      expect(instance.contracts.validators_set_contract).toHaveLength(66);
+      expect(instance.contracts.distribution_contract).toHaveLength(66);
+      expect(instance.contracts.rewards_manager_contract).toHaveLength(66);
+    }
+    let matches: number = 0;
+    for (const factory of context.contracts.factories) {
+      const factoryState = await factory.queryState();
+      for (const instance of res) {
+        if (instance.addr === factory.contractAddress) {
```

Comment:
you can simply compare two objects (state from locator vs state from factory itself), no need to check every contract record one by one, and in this case contract address length checking is no required. Try to have as less as possible logic in the tests.